### PR TITLE
Concurrent/Parallel user sharing

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds a sample API test for concurrent user sharing.
For now this test adds only a single share of resource to 5 of the existing user concurrently.

### Related Issue:
https://github.com/owncloud/QA/issues/828

### CODE REVIEW

**Comment by SagarGi** [comment]
#### Some Findings regarding the implementation of Concurrent user sharing using `cs3api`:
1. Currently concurrent sharing is done only for 5 users. It would have been good if we have more users to share concurrently. i am unable to find the request to create user through `cs3api` or also the available client for as says it is unimplemented
2. Also we need each sharee information to share. So to get the information of the created user we need to find each user information.

File: features/concurrent-user-sharing.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Feature: concurrent user sharing
+  As a user
+  I want to share their resource concurrently to different multiple users
+
+  @concurrent
```
Comment (v-scharf) [inline features/concurrent-user-sharing.feature]: Why do we need this tag? To quickly run the `go test --godog.tags=concurrent` test or do we need it in the future?

File: features/concurrent-user-sharing.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  Scenario: users make concurrent sharing to each other
+    Given user "admin" has logged in with password "admin"
+    And user "admin" has created a personal space with the alias "Admin Home"
+    And user "admin" has uploaded a file "testfile1.txt" with content "concurrent sharing" in the home directory with the alias "testfile1"
+    And user "admin" has uploaded a file "testfile2.txt" with content "concurrent sharing" in the home directory with the alias "testfile2"
```
Comment (v-scharf) [inline features/concurrent-user-sharing.feature]: unnecessary line

File: features/concurrent-user-sharing.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    When user "admin" shares a file "testfile1.txt" with the following users concurrently
```
Comment (v-scharf) [inline features/concurrent-user-sharing.feature]: I didn't find how create new users. I don't know if it's possible to do it here.
as a trick to increase a shares count can also be shared with demo groups

File: features/concurrent-user-sharing.feature (hunk 1 above)
Comment (SagarGi) [reply features/concurrent-user-sharing.feature]: oh sorry i forgot to remove it as this was part of local testing. Removing it.

File: features/concurrent-user-sharing.feature (hunk 3 above)
Comment (SagarGi) [reply features/concurrent-user-sharing.feature]: CC @butonic. If you could provide some details regarding the creation of the users.

**Comment by SagarGi** [comment]
Not very urgent thing to do right now. Can be done later. So moving to backlog.

File: features/concurrent-user-sharing.feature (hunk 3 above)
Comment (v-scharf) [reply features/concurrent-user-sharing.feature]: we cannot create more users. @SagarGi please add all existing group and we can merge PR

Comment (SagarGi) [reply features/concurrent-user-sharing.feature]: @ScharfViktor sure!

**Comment by SagarGi** [comment]
Since we cannot create more users with the `cs3api`. The main objective of this PR or issue was to include more users as much as possible and prepare the concurrency of the share for the created users. So concurrent sharing with only few users does not seem valid. May be we can achieve the thing with API tests in the ocis itself. So closing this PR.

**Pull request closed without merging**